crates/common/tedge_config/src/tedge_config_cli/tedge_config.rs

```diff
@@ -1353,11 +1353,12 @@ fn default_http_bind_address(dto: &TEdgeConfigDto) -> IpAddr {
 }
 
 fn device_id_from_cert(cert_path: &Utf8Path) -> Result<String, ReadError> {
-    let pem = PemCertificate::from_pem_file(cert_path)
-        .map_err(|err| cert_error_into_config_error(ReadableKey::DeviceId.to_cow_str(), err))?;
-    let device_id = pem
-        .subject_common_name()
-        .map_err(|err| cert_error_into_config_error(ReadableKey::DeviceId.to_cow_str(), err))?;
+    let pem = PemCertificate::from_pem_file(cert_path).map_err(|err| {
+        cert_error_into_config_error(ReadableKey::DeviceId.to_cow_str(), err, cert_path)
```

Comment:
This `device_id_from_cert` is called by all device id default functions (`device_id()`, `c8y_device_id()`, `az_device_id()`, and `aws_device_id()`. However, here gives `ReadableKey::DeviceId` as key. This is not 100% correct. It can be `ReadableKey::C8yDeviceId(None)`, `ReadableKey::AzDeviceId(Some("abc"))` and so on.

The difficulty is, the default functions cannot know which profile is used. That's why it is impossible to parse the precise ReadableKey to this fuction from the default device id functions.

Reply:
So the issue to improve the error message is that `device_id_from_cert` has no clue on the actual key. But can this key be passed as a parameter with the appropriate value by the caller (i.e device_id()`, `c8y_device_id()`, `az_device_id()`, and `aws_device_id()`) ?

Reply:
> So the issue to improve the error message is that `device_id_from_cert` has no clue on the actual key. 

Right.

> But can this key be passed as a parameter with the appropriate value by the caller (i.e `device_id()`, `c8y_device_id()`, `az_device_id()`, and `aws_device_id()`) ?

Also no unfortunately. Look at this function. This function is called by the macro. No chance to get the profile.
https://github.com/thin-edge/thin-edge.io/blob/91eb9bd53406f7b3e9d596ccef80e4719bc4953b/crates/common/tedge_config/src/tedge_config_cli/tedge_config.rs#L1375-L1383